### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by Markusplay**

### CODE REVIEW

File: src/components/common/layout/header/components/desktop-header/DesktopHeader.styles.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { SxProps, Theme } from '@mui/material/styles';
 import { alpha } from '@mui/system';
 
-import palette from '@/styles/theme/constants/pallete';
+import palette from '@/styles/theme/constants/palette';
```
Comment (andrewvitrenko) [inline src/components/common/layout/header/components/desktop-header/DesktopHeader.styles.ts:4]: replace with theme

File: src/components/common/layout/header/components/mobile-header/MobileHeader.styles.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { SxProps, Theme } from '@mui/material/styles';
 import { alpha } from '@mui/system';
 
-import palette from '@/styles/theme/constants/pallete';
+import palette from '@/styles/theme/constants/palette';
```
Comment (andrewvitrenko) [inline src/components/common/layout/header/components/mobile-header/MobileHeader.styles.ts:4]: replace with theme

File: src/components/common/ui/breadcrumbs-mui/Breadcrumbs.styles.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { SxProps, Theme } from '@mui/system';
 
-import palette from '@/styles/theme/constants/pallete/palette';
+import palette from '@/styles/theme/constants/palette/palette';
```
Comment (andrewvitrenko) [inline src/components/common/ui/breadcrumbs-mui/Breadcrumbs.styles.ts:3]: replace with theme

File: src/components/common/ui/cards/floating-card/FloatingCard.styles.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import { SxProps, Theme } from '@mui/material/styles';
-
-import palette from '@/styles/theme/constants/pallete';
+import palette from 'src/styles/theme/constants/palette';
```
Comment (andrewvitrenko) [inline src/components/common/ui/cards/floating-card/FloatingCard.styles.ts:2]: replace with `theme`

File: src/components/common/ui/column-chart/ColumnChart.styles.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { SxProps, Theme } from '@mui/material/styles';
 
-import palette from '@/styles/theme/constants/pallete';
+import palette from '@/styles/theme/constants/palette';
```
Comment (andrewvitrenko) [inline src/components/common/ui/column-chart/ColumnChart.styles.ts:3]: replace with `theme`

File: src/components/common/ui/column-chart/utils/data.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { ChartData } from 'chart.js';
 
 import { GetTeacherEvaluationsDTO } from '@/lib/api/teacher/dto/GetTeacherEvaluationsDTO';
-import palette from '@/styles/theme/constants/pallete';
+import palette from '@/styles/theme/constants/palette';
```
Comment (andrewvitrenko) [inline src/components/common/ui/column-chart/utils/data.ts:4]: replace with `theme`

File: src/components/common/ui/tab-mui/tab/Tab.styles.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { SxProps, Theme } from '@mui/material/styles';
 
-import palette from '@/styles/theme/constants/pallete';
+import palette from '@/styles/theme/constants/palette';
```
Comment (andrewvitrenko) [inline src/components/common/ui/tab-mui/tab/Tab.styles.ts:3]: replace with `theme`

File: src/components/common/ui/tab-mui/tab/Tab.styles.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
...
 import typography from '@/styles/theme/constants/typography';
```
Comment (andrewvitrenko) [inline src/components/common/ui/tab-mui/tab/Tab.styles.ts:4]: remove typography

File: src/components/pages/search-pages/subject-search/SubjectSearchList.tsx [hunk 9]
```diff
@@ -12,21 +12,24 @@ const TOAST_TIMER = 4000;
 export const SubjectSearchList = ({ subjects }: GetListOfSubjectsDTO) => {
   const router = useRouter();
   const toast = useToast();
+  const [subjectList, setSubjectList] = useState([]);
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchList.tsx:15]: add generic type

File: src/components/pages/search-pages/subject-search/SubjectSearchList.tsx [hunk 10]
```diff
@@ -12,21 +12,24 @@ const TOAST_TIMER = 4000;
...
 
   const redirect = (subjectId: string) => {
     router.push(`/subjects/${subjectId}/teachers`);
   };
 
   return (
     <ul className={styles['subject-search-list']}>
-      {subjects &&
-        subjects.map(subject => (
+      {subjectList &&
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchList.tsx:31]: remove condition, you always have an array in state

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 11]
```diff
@@ -43,7 +43,7 @@ const SubjectSearchPage = () => {
   const { data, isLoading, refetch, isFetching } =
     useQuery<GetListOfSubjectsDTO>(
       'subjects',
-      SubjectsAPI.getAll.bind(null, queryObj, pageSize * (curPage + 1)),
+      SubjectsAPI.getAll.bind(null, queryObj, PAGE_SIZE, curPage),
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:46]:
```suggestion
      () => SubjectsAPI.getAll(queryObj, PAGE_SIZE, curPage),
```

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 12]
```diff
@@ -28,7 +28,7 @@ const breadcrumbs = [
     href: '/subjects',
   },
 ];
-const pageSize = 20;
+const PAGE_SIZE = 20;
 
 const SubjectSearchPage = () => {
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:33]:
```suggestion
const SubjectSearchPage: FC = () => {
```

File: src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx [hunk 13]
```diff
@@ -28,7 +28,7 @@ const breadcrumbs = [
...
   const [queryObj, setQueryObj] = useState(SubjectInitialValues);
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/subject-search/SubjectSearchPage.tsx:34]: add generic type

File: src/components/pages/search-pages/teacher-search/TeacherSearchList.tsx [hunk 14]
```diff
@@ -18,22 +18,26 @@ export const TeacherSearchList = ({
   | (Omit<GetTeachersBySubjectDTO, 'subjectName'> & { className: string })) => {
   const toast = useToast();
 
+  const [teacherList, setTeacherList] = useState([]);
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/teacher-search/TeacherSearchList.tsx:21]: add generic type

File: src/components/pages/search-pages/teacher-search/TeacherSearchList.tsx [hunk 15]
```diff
@@ -18,22 +18,26 @@ export const TeacherSearchList = ({
...
     }
-  }, [teachers.length]);
+  }, [teachers]);
 
   return (
     <ul className={styles[`${className}-search-list`]}>
-      {teachers &&
-        teachers?.map((teacher, index) => (
-          <Link key={index} href={`/teachers/${teacher.id}`}>
+      {teacherList &&
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/teacher-search/TeacherSearchList.tsx:33]: remove condition

File: src/components/pages/search-pages/teacher-search/TeacherSearchPage.tsx [hunk 16]
```diff
@@ -44,7 +44,7 @@ export const TeacherSearchPage = () => {
 
   const { data, isLoading, refetch, isFetching } = useQuery<GetTeachersDTO>(
     'lecturers',
-    TeacherAPI.getAll.bind(null, queryObj, pageSize * (curPage + 1)),
+    TeacherAPI.getAll.bind(null, queryObj, PAGE_SIZE, curPage),
```
Comment (andrewvitrenko) [inline src/components/pages/search-pages/teacher-search/TeacherSearchPage.tsx:47]:
```suggestion
    () => TeacherAPI.getAll(queryObj, PAGE_SIZE, curPage),
```

File: src/lib/api/subject/SubjectAPI.ts [hunk 17]
```diff
@@ -26,12 +26,13 @@ export class SubjectsAPI {
   static async getAll(
     { search, order, sort, group }: SubjectSearchFormFields,
     pageSize: number,
+    page: number,
   ): Promise<GetListOfSubjectsDTO> {
     const url = `/subjects?${search ? `search=${search}` : ''}${
```
Comment (andrewvitrenko) [inline src/lib/api/subject/SubjectAPI.ts:31]: rewrite with axios params

File: src/lib/api/teacher/TeacherAPI.ts [hunk 18]
```diff
@@ -24,12 +24,13 @@ export class TeacherAPI {
   static async getAll(
     { search, order, sort, group }: TeacherSearchFormFields,
     pageSize: number,
+    page: number,
   ): Promise<{ teachers: GetTeacherDTO[] }> {
     const url = `/teachers?${search ? `search=${search}` : ''}${
```
Comment (andrewvitrenko) [inline src/lib/api/teacher/TeacherAPI.ts:29]: use axios params

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

**Comment by Markusplay** [comment]
due to huge amount of conflicts pull request will be done in new branch

**Pull request closed without merging**